### DISCUSSION TITLE
How to check weather request was replayed / received

### DISCUSSION THREAD

**Discussion by woozar** [mitmproxy/mitmproxy#5026] [Q&A]
I am using mitmproxy to mock the api for our ui tests. ​After a little bit of a learning curve, it works like a charm, so thank you!

We use Selenium tests in c# with dotnetcore2.1 where each test case starts a process that is running the proxy with the test cases own tape. There are some POST commands in the replays, that would trigger actions on the server and we want to make sure that those (or all) recorded calls where actually received (by the preoxy). 

Is there some way how to achieve this?

**Comment by woozar** [comment]
So am I the only one, who wants to make sure that a specific api call was mage during my replay :(

**Comment by mhils** [comment]
It's hard to solve this problem generically, but you can definitely write your own small addon that does some checks. It's really up to you how you want to consume the results. You could make your addon print to stdout, you could expose a small HTTP API, or you write a file, or you could exit with a specific exit code - all that should work. :)

**Reply by woozar** [reply]
The question was actually "what is already there" and not "what could I create" ;)
I didn't want to reinvent the wheel...
